packages/api/cms-api/src/warnings/warning-event-subscriber.ts

```diff
@@ -111,32 +111,30 @@ export class WarningEventSubscriber implements EventSubscriber {
             }
 
             const createWarnings = this.reflector.getAllAndOverride<CreateWarningsMeta>("createWarnings", [entity]);
-            if (createWarnings) {
-                const repository = this.entityManager.getRepository(entity);
+            if (createWarnings && args.changeSet?.originalEntity?.id) {
+                const repository: EntityRepository<{ id: string; scope: ContentScope }> = this.entityManager.getRepository(entity);
 
-                const rows = await repository.find();
+                const row = await repository.findOneOrFail(args.changeSet.originalEntity.id);
 
-                for (const row of rows) {
-                    let warnings: WarningData[] = [];
-                    if (this.isService(createWarnings)) {
-                        const service = this.moduleRef.get(createWarnings, { strict: false });
-                        warnings = await service.createWarnings(row);
-                    } else {
-                        warnings = await createWarnings(row);
-                    }
-                    const startDate = new Date();
-                    const sourceInfo = {
-                        rootEntityName: entity.name,
-                        rootPrimaryKey: args.meta.primaryKeys[0],
-                        targetId: row.id,
-                    };
-                    await this.warningService.saveWarnings({
-                        warnings,
-                        sourceInfo,
-                        scope: row.scope,
-                    });
-                    await this.warningService.deleteOutdatedWarnings({ date: startDate, sourceInfo });
+                let warnings: WarningData[] = [];
+                if (this.isService(createWarnings)) {
+                    const service = this.moduleRef.get(createWarnings, { strict: false });
+                    warnings = await service.createWarnings(row);
+                } else {
+                    warnings = await createWarnings(row);
                 }
+                const startDate = new Date();
+                const sourceInfo = {
+                    rootEntityName: entity.name,
+                    rootPrimaryKey: args.meta.primaryKeys[0],
+                    targetId: row.id,
```

Comment:
Another thing I realized: We are "fake primary key independent" because we save the primary key but everywhere else we just assume it to be id. 

I also did that now (since it's correct in 99% of the cases anyway), but we could think about changing that in a follow-up to really be primary key independent.